pkg/composer/serviceparser/serviceparser.go

```diff
@@ -500,8 +500,8 @@ func newContainer(project *types.Project, parsed *Service, i int) (*Container, e
 			c.RunArgs = append(c.RunArgs, fmt.Sprintf("-e=%s=%s", k, *v))
 		}
 	}
-	for _, v := range svc.ExtraHosts {
-		c.RunArgs = append(c.RunArgs, fmt.Sprintf("--add-host=%s", v))
+	for k, v := range svc.ExtraHosts {
+		c.RunArgs = append(c.RunArgs, fmt.Sprintf("--add-host=\"%s:%s\"", k, v))
```

Comment:
why do we need quotes?

Reply:
Because the colon should be escaped, sometimes it will be treated as an operator.

Reply:
What is an “operator” here

Reply:
https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion, I have tested it in my dev. I have many apps startup with `nerdctl compose`.

Reply:
I don’t think we are using bash here

Reply:
seems ok